src/Bicep.IO/Abstraction/IFileHandle.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bicep.IO.Abstraction
+{
+    public interface IFileHandle : IResourceHandle
+    {
+        IDirectoryHandle GetParent();
+
+        Stream OpenRead();
```

Comment:
@anthony-c-martin I looked into using `BinaryData` as the return type. While I appreciate its cleaner abstraction compared to Stream, I have some performance concerns since it always copies the data into a memory stream when created from a stream. I think it might be better to implement extension methods for reading and writing with `BinaryData`.